ValidationLibrary/Rules/HasNewestPtcsJenkinsLibRule.cs

```diff
@@ -31,31 +35,116 @@ public HasNewestPtcsJenkinsLibRule(ILogger logger)
+            // This method should be refactored when we have a better general idea how we want to fix things
+            var branchName = $"feature/{LibraryName}-update";
+            _logger.LogInformation("Rule {ruleClass} / {ruleName}, performing auto fix.", nameof(HasNewestPtcsJenkinsLibRule), RuleName);
+
+            var jenkinsContent = await GetJenkinsFileContent(client, repository);
+            if (jenkinsContent == null)
+            {
+                _logger.LogWarning("Rule {ruleClass} / {ruleName}, no {filename} found, unable to fix.",
+                    nameof(HasNewestPtcsJenkinsLibRule), RuleName, JenkinsFileName);
+                return;
+            }
+            var fixedContent = _regex.Replace(jenkinsContent.Content, $"'{LibraryName}@{_expectedVersion}'");
+
+            var branches = await client.Repository.Branch.GetAll(repository.Owner.Login, repository.Name);
+            if (branches.Any(branch => branch.Name == branchName))
+            {
+                _logger.LogInformation("Rule {ruleClass} / {ruleName}, Branch {branchName} already exists, skipping fix.",
+                     nameof(HasNewestPtcsJenkinsLibRule), RuleName, branchName);
+                return;
+            }
+            var branchReference = await client.Git.Reference.CreateBranch(repository.Owner.Login, repository.Name, branchName);
+            var master = await client.Git.Reference.Get(repository.Owner.Login, repository.Name, "heads/master");
+
+            var latest = await client.Git.Commit.Get(repository.Owner.Login, repository.Name, branchReference.Object.Sha);
+            _logger.LogTrace("Latest commit with message {a}", latest.Message);
+
+            var oldTree = await client.Git.Tree.Get(repository.Owner.Login, repository.Name, latest.Sha);
+            var newTree = new NewTree();
+            newTree.BaseTree = oldTree.Sha;
+
+            BlobReference blobReference = await CreateBlob(client, repository, fixedContent);
+
+            var treeItem = new NewTreeItem()
+            {
+                Path = JenkinsFileName,
+                Mode = FileMode,
+                Type = TreeType.Blob,
+                Sha = blobReference.Sha
+            };
+            newTree.Tree.Add(treeItem);
+
+            var createdTree = await client.Git.Tree.Create(repository.Owner.Login, repository.Name, newTree);
+            var commit = new NewCommit($"Updated {LibraryName} to latest.", createdTree.Sha, new[] { latest.Sha });
+            var commitResponse = await client.Git.Commit.Create(repository.Owner.Login, repository.Name, commit);
+
+            var refUpdate = new ReferenceUpdate(commitResponse.Sha);
+            await client.Git.Reference.Update(repository.Owner.Login, repository.Name, $"heads/{branchName}", refUpdate);
+
+            var pullRequest = new NewPullRequest($"[Automatic Validation] Updated {LibraryName} to latest.", branchReference.Ref, master.Ref);
```

Comment:
Same here
`[Automatic Validation] Update {LibraryName} to latest version`
If this is too long, skip the version part.

I base these on this article https://chris.beams.io/posts/git-commit/
I have started to really like those rules as a general guiding line for commit messages.

Reply:
Good poin!